sensor/common/networkflow/manager/purger.go

```diff
@@ -0,0 +1,305 @@
+package manager
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/pkg/centralsensor"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/pkg/timestamp"
+	"github.com/stackrox/rox/sensor/common"
+	"github.com/stackrox/rox/sensor/common/message"
+	flowMetrics "github.com/stackrox/rox/sensor/common/networkflow/metrics"
+)
+
+var (
+	// How often purger should run. The Purger removes old endpoints from activeEndpoints slice.
+	// This is important for cases when Collector or the orchestrator never reports a given endpoint
+	// as deleted, because there is no other mechanism that would remove an endpoint from memory.
+	purgerCycleSetting = env.EnrichmentPurgerTickerCycle.DurationSetting()
+)
+
+type PurgerOption func(purger *NetworkFlowPurger)
+
+// WithPurgerTicker overrides the default enrichment ticker
+func WithPurgerTicker(ticker <-chan time.Time) PurgerOption {
+	return func(purger *NetworkFlowPurger) {
+		if ticker != nil {
+			purger.purgerTickerC = ticker
+		}
+	}
+}
+
+type NetworkFlowPurger struct {
```

Comment:
This not a blocking comment but what do you think about making this a sub-component of the `networkFlowManager` instead of a `SensorComponent`? Similar to what we did for the [eventPipeline](https://github.com/stackrox/stackrox/blob/91d3ac047a4a491c5a92cb4e46d7e41651906194/sensor/kubernetes/eventpipeline/component/component.go#L9-L13). That way we could have an interface like:
```golang
type NetworkFlowComponent interface {
	Start() error
	Stop(error)
    Notify(common.SensorComponentEvent)
}
```

Reply:
Yeah, good point. Here is my take on that: [9a9603a](https://github.com/stackrox/stackrox/pull/14538/commits/9a9603a840cfb73e84d511b338d555122fc34a3e)